pkg/plugins/manager/pipeline/discovery/steps.go

```diff
@@ -17,33 +17,34 @@ func DefaultFindFunc(cfg *config.Cfg) FindFunc {
 	return finder.NewLocalFinder(cfg.DevMode, cfg.Features).Find
 }
 
-// DuplicatePluginValidation is a filter step that will filter out any plugins that are already registered with the
-// registry. This includes both the primary plugin and any child plugins, which are matched using the plugin ID field.
-type DuplicatePluginValidation struct {
+// DuplicatePluginVersionValidation is a filter step that will filter out any plugins that are already registered with the
+// registry. This includes both the primary plugin and any child plugins, which are matched using the plugin ID and
+// version fields.
+type DuplicatePluginVersionValidation struct {
 	registry registry.Service
 	log      log.Logger
 }
 
-// NewDuplicatePluginFilterStep returns a new DuplicatePluginValidation.
-func NewDuplicatePluginFilterStep(registry registry.Service) *DuplicatePluginValidation {
-	return &DuplicatePluginValidation{
+// NewDuplicatePluginVersionFilterStep returns a new DuplicatePluginVersionValidation.
+func NewDuplicatePluginVersionFilterStep(registry registry.Service) *DuplicatePluginVersionValidation {
```

Comment:
this is not used?

Reply:
No not currently. Was in two minds about this one since this should go hand in hand with how plugin management can now function with multiple plugins, which is why I left it (it will be used as part of the runner)  - but since it's not used internally here I could strip this out?

Reply:
Have removed this 👍